packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -113,6 +114,21 @@ export async function getCacheEntry(
   const cacheResult = response.result
   const cacheDownloadUrl = cacheResult?.archiveLocation
   if (!cacheDownloadUrl) {
+    // List cache for primary key only if cache miss occurs
+    const resource = `caches?key=${encodeURIComponent(keys[0])}`
+    const response = await httpClient.getJson<ArtifactCacheList>(getCacheApiUrl(resource))
```

Comment:
So we are using AC endpoint directly and not GitHub REST API. I feel it should be fine but tagging @anuragc617 for his review.

Reply:
Other functions were doing the same so I thought should be fine. Little unsure about permissions need for such call to AC.